src/EditorFeatures/DiagnosticsTestUtilities/CodeActions/AbstractCodeActionOrUserDiagnosticTest.cs

```diff
@@ -803,5 +800,89 @@ protected static (OptionKey, object) SingleOption<T>(PerLanguageOption<CodeStyle
 
             await TestActionCountAsync(input, outputs.Length, parameters);
         }
+
+#if !CODE_STYLE
+        internal interface IOptionsCollection : IDictionary<OptionKey2, object>
```

Comment:
❔ Can we not use this interface definition in both layers to avoid some of the `#if` lines?

Reply:
Yes, possibly - I tried not to worry too much about clean design here as hopefully we can delete this entire legacy framework file soon.